mailer.js

```diff
@@ -456,12 +472,32 @@ module.exports = function (log) {
         link: link,
         oneClickLink: oneClickLink,
         supportUrl: this.supportUrl,
-        supportLinkAttributes: this._supportLinkAttributes()
+        supportLinkAttributes: this._supportLinkAttributes(template)
       },
       uid: message.uid
     })
   }
 
+  Mailer.prototype._generateUTMLink = function (link, queryParams, template, context) {
+    if (!queryParams) {
+      queryParams = {}
+    }
+
+    queryParams['utm_source'] = 'email'
+    queryParams['utm_medium'] = 'email'
+
+    var campaign = utmCampaignMap[template]
+    if (campaign) {
+      queryParams['utm_campaign'] = utmPrefix + campaign
+    }
+
+    if (context) {
```

Comment:
For the idea above about fetching context from a map, it could be implemented here identically to how the campaign is fetched from `utmCampaignMap`.

```js
var context = utmContextMap[template]
if (context) {
  queryParams['utm_context'] = utmPrefix + context + contextSuffix
}
```